src/font.rs

```diff
@@ -105,15 +103,271 @@ struct Loca<'a>(&'a [u8]);
+            .field("offset", &self.get_offset())
+            .finish()
+    }
+}
+
+struct Encoding<'a>(&'a [u8]);
+
+impl<'a> Encoding<'a> {
+    fn get_format(&'a self) -> u16 {
+        get_u16(self.0, 0).unwrap()
+    }
+
+    fn get_length(&'a self) -> u16 {
+        get_u16(self.0, 2).unwrap()
+    }
+
+    fn get_language(&'a self) -> u16 {
+        get_u16(self.0, 4).unwrap()
+    }
+
+    fn get_seg_count_x_2(&'a self) -> u16 {
+        get_u16(self.0, 6).unwrap()
+    }
+
+    fn get_seg_count(&'a self) -> u16 {
+        self.get_seg_count_x_2() / 2
+    }
+
+    fn get_search_range(&'a self) -> u16 {
+        get_u16(self.0, 8).unwrap()
+    }
+
+    fn get_entry_selector(&'a self) -> u16 {
+        get_u16(self.0, 10).unwrap()
+    }
+
+    fn get_range_shift(&'a self) -> u16 {
+        get_u16(self.0, 12).unwrap()
+    }
+
+    fn get_u16_vec(&'a self, start_position: u16, count: u16) -> Vec<u16> {
+        let mut result = vec![];
+        let mut vec_position = start_position;
+        let limit = vec_position + 2 * count;
+        while vec_position < limit {
+            result.push(get_u16(self.0, vec_position as usize).unwrap());
+            vec_position += 2;
+        }
+        result
+    }
+
+    fn get_i16_vec(&'a self, start_position: u16, count: u16) -> Vec<i16> {
+        let mut result = vec![];
+        let mut vec_position = start_position;
+        let limit = vec_position + 2 * count;
+        while vec_position < limit {
+            result.push(get_i16(self.0, vec_position as usize).unwrap());
+            vec_position += 2;
+        }
+        result
+    }
+
+    fn get_end_counts_position() -> u16 {
+        14
+    }
+
+    fn get_end_counts(&'a self) -> Vec<u16> {
+        let seg_count = self.get_seg_count();
+        self.get_u16_vec(Self::get_end_counts_position(), seg_count)
+    }
+
+    fn get_start_counts_position(seg_count: u16) -> u16 {
+        Self::get_end_counts_position() + 2 + 2 * seg_count
+    }
+
+    fn get_start_counts(&'a self) -> Vec<u16> {
+        let seg_count = self.get_seg_count();
+        self.get_u16_vec(Self::get_start_counts_position(seg_count), seg_count)
+    }
+
+    fn get_id_deltas_position(seg_count: u16) -> u16 {
+        Self::get_start_counts_position(seg_count) + 2 * seg_count
+    }
+
+    fn get_id_deltas(&'a self) -> Vec<i16> {
+        let seg_count = self.get_seg_count();
+        self.get_i16_vec(Self::get_id_deltas_position(seg_count), seg_count)
+    }
+
+    fn get_id_range_offset_position(seg_count: u16) -> u16 {
+        Self::get_id_deltas_position(seg_count) + 2 * seg_count
+    }
+
+    fn get_id_range_offsets(&'a self) -> Vec<u16> {
+        let seg_count = self.get_seg_count();
+        self.get_u16_vec(Self::get_id_range_offset_position(seg_count), seg_count)
+    }
+
+    fn extract_glyph_id(
+        &'a self, code_point: u16, start_value: u16, seg_count: u16, seg_index: u16,
+    ) -> Option<u16> {
+        let data = &self.0;
```

Comment:
This can be just `self.0`. That's my fault, I led you astray (it works with the extra ref because magic).

Reply:
I think I just flubbed this edit all on my own. 😀